devenv/lib/archive.py

```diff
@@ -20,17 +22,34 @@ def atomic_replace(src: str, dest: str) -> None:
     os.replace(src, dest)
 
 
-def download(url: str, sha256: str, dest: str = "") -> str:
+def download(
```

Comment:
should i add some unittests for this? integration won't exercise retries generally

Reply:
will be adding a unittest and separate pr